src/pages/test-report/components/reverse-left/reverse-left.ts

```diff
@@ -0,0 +1,148 @@
+    private testDataByCategory: TestDataByCategoryProvider,
+    private manoeuvresByCategory: ManoeuvresByCategoryProvider,
+  ) {}
+
+  ngOnInit(): void {
+    this.reducerInfo = this.getReducerInfo(this.testCategory);
+    const manoeuvres$ = this.store$.pipe(
+        select(getTests),
+        select(getCurrentTest),
+        select(this.testDataByCategory.getTestDataByCategoryCode(this.testCategory)),
+        select(this.manoeuvresByCategory.getManoeuvresByCategoryCode(this.testCategory)),
+      );
+
+    this.subscription = manoeuvres$.subscribe((manoeuvres: ManoeuvreUnion) => {
+      this.drivingFaults =
+          this.faultCountProvider.getManoeuvreFaultCount(this.testCategory, manoeuvres, CompetencyOutcome.DF);
+      this.hasSeriousFault =
+          this.faultCountProvider.getManoeuvreFaultCount(this.testCategory, manoeuvres, CompetencyOutcome.S) > 0;
+      this.hasDangerousFault =
+          this.faultCountProvider.getManoeuvreFaultCount(this.testCategory, manoeuvres, CompetencyOutcome.D) > 0;
+      this.completedReverseLeft = this.reducerInfo.completedReverseLeft.call(this, manoeuvres);
+    });
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  toggleReverseLeft = (): void => {
+    if (this.completedReverseLeft && !this.hasFaults()) {
+      this.store$.dispatch(this.reducerInfo.deselectReverseLeftManoeuvre);
+      return;
+    }
+    this.store$.dispatch(new RecordManoeuvresSelection(ManoeuvreTypes.reverseLeft));
+  }
+
+  hasFaults = (): boolean => {
+    return this.drivingFaults > 0 || this.hasSeriousFault || this.hasDangerousFault;
+  }
+
+  togglePopoverDisplay = (): void => {
+    if (this.displayPopover) {
+      this.store$.dispatch(new ReverseLeftPopoverClosed());
+      this.displayPopover = false;
+    } else {
+      this.store$.dispatch(new ReverseLeftPopoverOpened());
+      this.displayPopover = true;
+    }
+    this.toggleOverlay();
+  }
+
+  toggleOverlay(): void {
+    if (this.clickCallback) {
+      this.clickCallback.callbackMethod();
+    }
+  }
+
+  getReducerInfo(categoryCode: CategoryCode) {
+    const testCategory = categoryCode as TestCategory;
+    switch (testCategory) {
+      case TestCategory.C:
+      case TestCategory.C1:
+      case TestCategory.C1E:
+      case TestCategory.CE:
+        return {
+          completedReverseLeft: getReverseLeftSelectedC,
+          deselectReverseLeftManoeuvre: new DeselectReverseLeftManoeuvreCatC(),
+        };
+      case TestCategory.D1:
+      case TestCategory.D1E:
+      case TestCategory.D:
+      case TestCategory.DE:
+        return {
+          completedReverseLeft: getReverseLeftSelectedD,
+          deselectReverseLeftManoeuvre: new DeselectReverseLeftManoeuvreCatD(),
+        };
+      default:
+        return {
+          completedReverseLeft: getReverseLeftSelectedBE,
+          deselectReverseLeftManoeuvre: new DeselectReverseLeftManoeuvre(),
+        };
+    }
```

Comment:
Can we only have 1 common set of actions used across all of the reducers. We do something similar for Uncouple Recouple

Reply:
This will be addressed as a part of a separate story: https://jira.dvsacloud.uk/browse/MES-4854